### ISSUE TITLE
Input parameters for  Reducers

### ISSUE THREAD

**Issue by AdieOlami** [ReSwift/ReSwift#406]
I am using this library for the first time and it is looking interesting, but I have an issue when making a networking call that takes input parameter like authentication

below is an example of what I have tried

```
struct AuthReducer {
    
    let networkAdapter: NetworkAdapter
    let disposeBag = DisposeBag()
    
    func authReducer(action: Action, state: LoginState?) -> LoginState {
        
        let state = state ?? LoginState(login: .loading)
        switch action {
        case _ as LoginAction:
            _ = networkAdapter.login(phoneOrEmail: <#T##String#>, password: <#T##String#>)
                .subscribe({(event) in
                    switch event {
                        
                    case .next(let response):
                        store.dispatch(SetUserAction(user: response.user!))
                    case .error(let error):
                        store.dispatch(SetErrorAction(error: error))
                    case .completed:
                        break
                    }
                }).disposed(by: disposeBag)
        default: break
        }
        
        return state
    }
}

the phone or email field are user inputs, I can I achieve this. it is not clearly stated in the docs
```

**Comment by DivineDominion** [comment]
Try to make Reducers independent from services of your app. They get an action and a state, and these two parameters should contain all the info they need.

The _result_ of network request is a candidate for a good action. The request itself isn't.

Async Actions are now discouraged and we recommend to use Thunks: https://github.com/ReSwift/ReSwift-Thunk/

The Thunk middleware's Readme contains an example you might be interested in :)

**Comment by AdieOlami** [comment]
Would surely check it out.


I tried using ReSwiftRouter and I got the following error. Any help as to why?

```
[!] CocoaPods could not find compatible versions for pod "ReSwift":
  In snapshot (Podfile.lock):
    ReSwift (= 4.1.1)

  In Podfile:
    ReSwift

    ReSwiftRouter (from `https://github.com/ReSwift/ReSwift-Router.git`) was resolved to 0.6.0, which depends on
      ReSwift (~> 4.0.0)

Specs satisfying the `ReSwift, ReSwift (= 4.1.1), ReSwift (~> 4.0.0)` dependency were found, but they required a higher minimum deployment target.
```

**Comment by DivineDominion** [comment]
Not sure, as the Router package is mostly unmaintained and dormant. Noticed you opened ReSwift/ReSwift-Router#114 already, so I'll close this one for now
